homeassistant/components/logbook/__init__.py

```diff
@@ -583,28 +583,40 @@ def augment(self, data: dict[str, Any], entity_id: str | None, row: Row) -> None
             data["context_event_type"] = event_type
             return
 
-        if not entity_id:
+        if event_type in self.external_events:
+            domain, describe_event = self.external_events[event_type]
+            data["context_event_type"] = event_type
+            data["context_domain"] = domain
+            event = self.event_cache.get(context_row)
+            described = describe_event(event)
+            if name := described.get(ATTR_NAME):
+                data["context_name"] = name
+            if attr_entity_id := described.get(ATTR_ENTITY_ID):
+                data["context_entity_id"] = attr_entity_id
+                data["context_entity_id_name"] = self.entity_name_cache.get(
+                    attr_entity_id, context_row
+                )
+            if ATTR_MESSAGE in described:
+                data["context_message"] = described[ATTR_MESSAGE]
             return
 
-        attr_entity_id = _row_event_data_extract(context_row, ENTITY_ID_JSON_EXTRACT)
-        if attr_entity_id is None or (
-            event_type in SCRIPT_AUTOMATION_EVENTS and attr_entity_id == entity_id
+        if (
```

Comment:
So the only time we actually would hit this code path is for `EVENT_LOGBOOK_ENTRY` since its the only one that isn't going to be in `self.external_events`

Reply:
I guess someone could write an automation that when a logbook entry is added for X do Y

Reply:
That seems almost crazy to support though.

I should just delete this code.